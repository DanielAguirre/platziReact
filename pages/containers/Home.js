import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Post from  '../../posts/containers/post.js';
import Loading from '../../shared/components/Loading';
import api from  '../../api';
import style from './Page.css';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 1,
      posts: [],
      loading: true,
    };
  }
  async componentDidMount() {
    const posts = await api.posts.getList(this.state.page);
    this.setState({
      posts,
      page: this.state.page + 1,
      loading: false,
    });
    this.handleScroll = this.handleScroll.bind(this);

    window.addEventListener('scroll', this.handleScroll);
  }

  handleScroll (event) {
    if (this.state.loading) return null;

    const scrolled = window.scrollY;
    const viewportHeight = window.innerHeight;
    const fullHeight = document.documentElement.clientHeight;

    if (!(scrolled + viewportHeight + 300 >= fullHeight)){
      return null;
    }

    this.setState({ loading:true }, async () => {
      try {
        const post = await api.posts.getList(this.state.page);
        this.setState({
          posts: this.state.posts.concat(post),
          page: this.state.page +1,
          loading: false,
        });
      }catch(error) {
        console.error(error);
        this.setState({
          loading: true,
        });
      }
    });
  }
  render() {
    return (
      <section name="Home" className="section">
        <section className="list">
          { this.state.loading && (
            <Loading />
          )}
          {this.state.posts
            .map(post => <Post key={post.id} {...post} />)}
        </section>
        <Link to="/about">Go to about</Link>
      </section>
    );
  }
}

export default Home;
