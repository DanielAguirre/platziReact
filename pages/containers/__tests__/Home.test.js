import React from 'react';
import { shallow } from 'enzyme';
import Home from '../Home';

it('Render the add componet', () => {
  const componet = shallow(<Home/>);
  expect(componet).toMatchSnapshot();
});