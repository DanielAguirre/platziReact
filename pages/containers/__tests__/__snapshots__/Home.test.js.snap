// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the add componet 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    name="Home"
>
    <h1>
        Home
    </h1>
    <section>
        <h2>
            Loading posts...
        </h2>
    </section>
    <Link
        replace={false}
        to="/about"
    >
        Go to about
    </Link>
</section>,
  "nodes": Array [
    <section
      name="Home"
>
      <h1>
            Home
      </h1>
      <section>
            <h2>
                  Loading posts...
            </h2>
      </section>
      <Link
            replace={false}
            to="/about"
      >
            Go to about
      </Link>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "loading": true,
          "page": 1,
          "posts": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          name="Home"
>
          <h1>
                    Home
          </h1>
          <section>
                    <h2>
                              Loading posts...
                    </h2>
          </section>
          <Link
                    replace={false}
                    to="/about"
          >
                    Go to about
          </Link>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          name="Home"
>
          <h1>
                    Home
          </h1>
          <section>
                    <h2>
                              Loading posts...
                    </h2>
          </section>
          <Link
                    replace={false}
                    to="/about"
          >
                    Go to about
          </Link>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home />,
}
`;
